hikari/iterators.py

```diff
@@ -815,6 +815,18 @@ async def __anext__(self) -> ValueT:
             self._complete()
 
 
+class NOOPLazyIterator(typing.Generic[ValueT], LazyIterator[ValueT]):
```

Comment:
Instead of a NOOPLazyIterator, LazyIterator could be a no-op by default instead.

I have already been thinking about splitting async LazyIterators from sync LazyIterators which would help with cases like the CacheView iterator needing to be awaited even if it is sync.

This would be part of a greater revamp of the LazyIterators, so might be worth waiting for that to be done (by me or someone else interested in taking a look) instead of half-doing it here.